example.init.yaml

```diff
@@ -2,36 +2,48 @@ orchestration: kubernetes
 secrets:
   ### Generated secrets (do not edit)
   - name: s3-secret-key
-    namespace: "openfaas-fn"
     literals:
       - name: s3-secret-key
+    filters:
+      - "default"
```

Comment:
Do we need an explicit filter of `default` when these are non-optional? (Tell me a bit about the design/approach)

Reply:
If you look the code this `plan = filterFeatures(plan)` is the place I check the struct and append the features according to what is populated. I always add `default` filter to the features so when filtering secrets, the mandatory ones are always created. That way the filtering of secrets is easier with 1 function which just checks if the `filters` field of the secret contains one of the available features which are enabled/configured in the `.yaml`. In case you want secret created, for whatever reason, create your secret and add `default` filter and it will be created. I will document that if neccessary.